compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1500,13 +1509,21 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
           if (!param.tpt.isEmpty) param
           else
             val formal = protoFormal(i)
+            val isWildcardParam = areWildcardParams.map(list => if i < list.length then list(i) else false).getOrElse(false)
             val knownFormal = isFullyDefined(formal, ForceDegree.failBottom)
-            val paramType =
-              if knownFormal then formal
-              else inferredFromTarget(param, formal, calleeType, paramIndex)
-                .orElse(errorType(AnonymousFunctionMissingParamType(param, tree, formal), param.srcPos))
+            // Since decomposeProtoFunction eagerly approximates function arguments
+            // from below, then in the case that the argument was also identified as
+            // a wildcard type we try to prioritize inferring from target, if possible.
+            // See issue 16405 (tests/run/16405.scala)
+            val (usingFormal, paramType) =
```

Comment:
Not sure why usingFormal is needed here. Why is`knownFormal` not the right condition?